server/controllers/postLogic.js

```diff
@@ -4,20 +4,29 @@ const User = mongoose.model('User');
 
 exports.getPost = async (req, res) => {
   let response;
-  if (req.params.id) {
-    response = await Post.findById(req.params.id).populate({
-      path: 'author',
-      select: ['email', 'first', 'last']
+
+  if (req.query.postId) { // Gets post by post Id
+    response = await Post.findById(req.query.postId).populate({ path: 'author', select: ['email', 'first', 'last'] });
+  } else if (req.query.userId) { // Gets all posts by user Id
+    await Post.find({}, (err, posts) => {
+      let userPosts = [];
+
+      posts.forEach(post => {
```

Comment:
@ShawnToubeau I might refactor this later down the line for latency purposes. When you are retrieving the posts of a userId, the most efficient way would be to go to the user document of that id, then return their posts parameter, which is an array of post ObjectID's, then populate those object ID's with the post model like on line 9 did with the user model.